docs/docs/components/runtime.md

```diff
@@ -42,17 +42,25 @@ The `handler` field in the Kubernetes `RuntimeClass` instructs containerd not to
 Instead, containerd invokes a custom plugin called `containerd-shim-contrast-cc-v2`.
 This shim is described in more detail in the [upstream source repository](https://github.com/kata-containers/kata-containers/tree/3.4.0/src/runtime) and in the [containerd documentation](https://github.com/containerd/containerd/blob/main/core/runtime/v2/README.md).
 
-### `cloud-hypervisor` virtual machine manager (VMM)
+### Virtual machine manager (VMM)
 
-The `containerd` shim uses [`cloud-hypervisor`](https://www.cloudhypervisor.org) to create a confidential virtual machine for every pod.
-This requires the `cloud-hypervisor` binary to be installed on every node (responsibility of the [`node-installer`](#node-installer-daemonset)).
+The `containerd` shim uses a virtual machine monitor to create a confidential virtual machine for every pod.
+On AKS, Contrast uses [`cloud-hypervisor`](https://www.cloudhypervisor.org).
+On bare metal, Contrast uses [`QEMU`](https://www.qemu.org/).
+The appropriate files need to be installed on every node (which is the responsibility of the [`node-installer`](#node-installer-daemonset)).
 
-### `Tardev snapshotter`
+### Snapshotters
 
-Contrast uses a special [`containerd` snapshotter](https://github.com/containerd/containerd/tree/v1.7.16/docs/snapshotters/README.md) ([`tardev`](https://github.com/kata-containers/tardev-snapshotter)) to provide container images as block devices to the pod-VM. This snapshotter consists of a host component that pulls container images and a guest component (kernel module) used to mount container images.
+Contrast uses special [`containerd` snapshotters](https://github.com/containerd/containerd/tree/v1.7.16/docs/snapshotters/README.md) to provide container images to the pod-VM.
+Each snapshotter consists of a host component that pulls container images and a guest component used to mount/pull container images.
+
+On AKS, Contrast uses the [`tardev`](https://github.com/kata-containers/tardev-snapshotter) snapshotter to provide container images as block devices to the pod-VM.
 The `tardev` snapshotter uses [`dm-verity`](https://docs.kernel.org/admin-guide/device-mapper/verity.html) to protect the integrity of container images.
 Expected `dm-verity` container image hashes are part of Contrast runtime policies and are enforced by the kata-agent.
-This enables workload attestation by specifying the allowed container image as part of the policy. Read [the chapter on policies](policies.md) for more information.
+
+On bare metal, Contrast uses the [`nydus`](https://nydus.dev/) snapshotter to allow the guest to pull container images securely.
+
+In all cases, hashes of the container images are stored in the Contrast runtime policies to enable workload attestation. Read [the chapter on policies](policies.md) for more information.
```

Comment:
Is that true? I don't think we implemented that for nydus yet.

Reply:
I was under the impression that the guest component that pulls the image would use the hash from the `io.kubernetes.cri.image-name` annotation, and that that's why @burgerdev fixed those libraries that didn't actually enforce the hash.

![image](https://github.com/user-attachments/assets/9f7cf3b2-8d39-4110-9c76-9dfb65616bf0)

Is this wrong?

Reply:
Hm, no, this can't possibly be the right hash because nydus uses a completely different hashing scheme (it kind of needs to because of all the lazy loading that makes it impossible to compute a hash over the entire data).

Reply:
There is no lazy-loading involved. The hash is would be verified, but we didn't pull in the distribution patch, and we also don't have any enforcement yet that images must have a digest.

Reply:
> There is no lazy-loading involved.

Why not? AFAICT, the FUSE daemon running inside the pod VM checks the integrity of the content lazily as it is accessed.

> The hash is would be verified, but we didn't pull in the distribution patch,

What patch are you referring to?

> and we also don't have any enforcement yet that images must have a digest.

Right, I was under the impression that we did because I always saw hashes in the deployments during testing, but that's just an artifact of the image replacements file.

> And we are still using allow-all policy.

Sure, but we won't by the time we release bare metal.